maas/client/viscera/tests/test_zones.py

```diff
@@ -0,0 +1,81 @@
+"""Tests for `maas.client.viscera.zones`."""
+
+import random
+
+from testtools.matchers import Equals
+
+from .. import zones
+
+from ..testing import bind
+from ...testing import (
+    make_string_without_spaces,
+    TestCase,
+)
+
+
+def make_origin():
+    """
+    Create a new origin with Zone and Zones. The former
+    refers to the latter via the origin, hence why it must be bound.
+    """
+    return bind(zones.Zones, zones.Zone)
+
+
+class TestZones(TestCase):
+
+    def _assert_create_zone(self, **kwargs):
+        Zones = make_origin().Zones
+        zone = {"id": random.randint(0, 100)}
+        zone.update(kwargs)
+        Zones._handler.create.return_value = zone
+        Zones.create(**kwargs)
+        Zones._handler.create.assert_called_once_with(**kwargs)
```

Comment:
The indirection of this method obscures the purpose of each test that uses it, and in the end saves only a little repetition. It also doesn't show one of the most important things: that a new `Zone` is returned, and that it has been populated correctly.

For example:

```python
def test__zones_create(self):
    origin = make_origin()
    zone_id = random.randint(0, 100)
    name = make_string_without_spaces()
    desc = make_string()
    origin.Zones._handler.create.return_value = {
        "id": zone_id,
        "name": name,
        "description": desc,
    }
    zone = origin.Zones.create(name=name, desc=desc)
    self.assertThat(zone, IsInstance(origin.Zone))
    self.assertThat(zone, MatchesStructure.byEquality(
        id=zone_id, name=name, description=desc))
```

For the other test, an interesting thing to check is that the `Zone` works correctly when the `description` field is either missing or null or the empty string in the data returned from the server (whichever it is in practice; I don't know).